hooks/use-summary.ts

```diff
@@ -0,0 +1,38 @@
+import { ChatContext } from '@/app/chat-context';
+import { ChatHistoryResponse } from '@/types/chat';
+import { useCallback, useContext } from 'react';
+import useChat from './use-chat';
```

Comment:
maybe we can design a universal stream request hook. `use-summary` and `use-chat` have a lot of code duplication.